src/main/java/seedu/mtracker/MTracker.java

```diff
@@ -14,24 +16,33 @@ public MTracker() {
     }
 
     public void run() {
-        Command command;
-        TextUi.greetAtStartUp();
-        String userInput = InputParser.getUserInput();
-        String[] commandComponents = InputParser.getCommandComponents(userInput);
-        try {
-            command = InputParser.filterByCommandType(commandComponents);
-            command.execute();
-        } catch (Exception e) {
-            TextUi.showErrorMessage(e);
+        Command command = new InvalidCommand();
```

Comment:
Try not to initialise command to invalid at the start, it works in the program wise but its confusing to the reader
https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#don-t-recycle-variables-or-parameters

Rather initialise command in the try catch block. I know you have a while loop so use a do while loop instead and in the catch block initialise the command as invalidCommand

Reply:
OK that is smart.